src/main/java/app/hero_maker/DataValidator.java

```diff
@@ -0,0 +1,33 @@
+package app.hero_maker;
+
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+
+import java.util.Arrays;
+
+@Component
+public class DataValidator implements org.springframework.validation.Validator {
+
+    public boolean supports(Class clazz) {
+        return Hero.class.equals(clazz);
+    }
+
+    @Override
+    public void validate(Object obj, Errors error) {
+        ValidationUtils.rejectIfEmptyOrWhitespace(error, "name", "Name is empty.");
```

Comment:
nie rob magicznych stringow, wydziel do zmiennej, mozesz tez zrobic interface na takie ogolne nazwy np 
interfejs ErrorsKeys i dac tam takie ogolne nazwy jak "FIELD_EMPTY", to samo z intami, zrob zmienna np MAX_STAT albo jak ja tam nazwiesz